src/data_cleaning.py

```diff
@@ -1457,15 +1467,57 @@ def identify_partial_cems_plants(all_data):
         validate="m:1",
     )
 
-    # for plants where the data source is eia only, and it exists in partial plant, change the source to partial_cems_plant
+    # Plants with these primary fuel types shouldn't be shaped using CEMS data --
+    # any cems data is likely to be a backup generator and therefore not representative of overall generation.
+    non_cems_fuels = CLEAN_FUELS + ["GEO"]
+
+    # for plants where the data source is eia only, and it exists in partial plant, 
+    # change the source to partial_cems_plant
+    # but only if the fuel type is consistent with CEMS fuel types
     all_data.loc[
         (all_data["hourly_data_source"] == "eia")
-        & (all_data["partial_plant"] == "both"),
+        & (all_data["partial_plant"] == "both")
+        & (
+            ~all_data["energy_source_code"].isin(non_cems_fuels)
+        ),  # CEMS data should only be used to shape if plant is primarilly fossil
         "hourly_data_source",
     ] = "partial_cems_plant"
 
+    # Because we exclude clean generators from the partial_cems_plant method, it
+    # is possible that data for a single plant will end up in two different 
+    # output files. Thus, we want to identify if there are any plants that have
+    # a mix of methods.
+    problem_subplant_months = (
```

Comment:
I just want to confirm: are we trying to prevent mixed methods within a single subplant, or mixed methods within a single plant? This check is not trying to prevent, for example, a nuclear subplant and a diesel subplant from ending up in two different files, only the case where for some reason a nuclear generator and diesel generator were lumped into the same subplant?

Reply:
I also think we should maybe update this variable name to be slightly more descriptive, eg `subplant_month_with_mixed_hourly_methods`

Reply:
this check prevents mixed methods within a single subplant. It's functionally a validation check for the mixed NaN-id subplants that were fixed by your subplant_id PR

Reply:
clarified comment and var name in [84e4a9e](https://github.com/singularity-energy/open-grid-emissions/pull/238/commits/84e4a9ebf9d56a0f69684ac876a54c3cbb11880b)